File: services/src/main/java/org/auditioner/services/family/Family.java
package org.auditioner.services.family;

import org.auditioner.services.family.member.FamilyMember;

public class Family {
    private String location;
    private String name;
    private String email;
    private String phone;
    private String address01;
    private String address02;
    private String city;
    private String state;
    private String zipCode;
    private String preferredContactMethod;

    public String getPreferredContactMethod() { return preferredContactMethod; }

    public void setPreferredContactMethod(String preferredContactMethod) {
        this.preferredContactMethod = preferredContactMethod;
    }

    public String getEmail() { return email; }

    public void setEmail(String email) { this.email = email; }

    public String getPhone() { return phone; }

    public void setPhone(String phone) { this.phone = phone; }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public String getAddress01() {
        return address01;
    }

    public void setAddress01(String address01) {
        this.address01 = address01;
    }

    public String getAddress02() {
        return address02;
    }

    public void setAddress02(String address02) {
        this.address02 = address02;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public String getZipCode() {
        return zipCode;
    }

    public void setZipCode(String zipCode) {
        this.zipCode = zipCode;
    }
}

File: web/app/models/family.js
import Ember from 'ember';

export default Ember.Object.extend({
  path: null,
  name: null,
  loadFromData: function (client, data) {
    this.set('client', client);

    this.set('id', data.location.substring(data.location.lastIndexOf("/") + 1));
    this.set('location', data.location);
    this.set('name', data.name);
    this.set('email', data.email);

    let phone = data.phone;
    if (phone && phone.length === 10) {
      phone = "(" + phone.substr(0, 3) + ") " + phone.substr(3, 3) + "-" + phone.substr(6, 4);
    }

    this.set('phone', phone);
    this.set('address01', data.address01);
    this.set('address02', data.address02);
    this.set('city', data.city);
    this.set('state', data.state);
    this.set('zipCode', data.zipCode);
    this.set('preferredContactMethod', data.preferredContactMethod);


  },
  toJson: function () {

    let phone = this.get('phone');
    console.log(phone);
    if (phone) {
      phone = phone.replace(/[^0-9.]/g, "").substr(0, 10);
    }
    console.log(phone);

    return JSON.stringify({
      name: this.get('name'),
      email: this.get('email'),
      phone: phone,
      address01: this.get('address01'),
      address02: this.get('address02'),
      city: this.get('city'),
      state: this.get('state'),
      zipCode: this.get('zipCode'),
      preferredContactMethod: this.get('preferredContactMethod'),

    });
  }
});
